src/sonic-config-engine/tests/common_utils.py

```diff
@@ -0,0 +1,33 @@
+import json
+import re
+import sys
+
+PY3x = sys.version_info >= (3, 0)
+PYvX_DIR = "py3" if PY3x else "py2"
+PYTHON_INTERPRETTER = "python3" if PY3x else "python2"
+
+def tuple_to_str(tuplestr):
+    """ Convert Python tuple '('elem1', 'elem2')' representation into string on the for "elem1|elem2" """
+    def to_str(tupleobj):
+        tupleobj = re.sub(r"([\(\)])", '"', tupleobj.group(0))
+        return re.sub(r"( *, *)", '|', tupleobj).replace("'", '')
+
+    return re.sub(r"(\(.*?\))", to_str, tuplestr)
+
+def to_dict(dictstr):
+    """ Convert string represention of Python dict into dict """
+    # handle tuple elements
+    dictstr = tuple_to_str(dictstr)
+
+    return json.loads(dictstr.replace("'", '"'))
```

Comment:
will python `eval()` fucntion does the job?

Reply:
No, it ill not as tuple were also failing when using AssertEqual from pytest.